src/shared/helpers/getActualDataList.ts

```diff
@@ -1,3 +1,4 @@
+import { getCourseDate } from './getCourseDate';
 import { dayJS } from '@/app/services/dayjs';
 
 type DataListType<T> = Array<T>;
```

Comment:
Do we really need this type?

Reply:
And I suggest to restrict `DataItemWithDate ` to remove comments:
```
export type DateType = `${string & { __regexPattern: '\\d{4}-\\d{2}-\\d{2}' }}`;

interface DataItemWithDate extends DataItem {
  date: DateType; // 'YYYY-MM-DD'
}
```

Reply:
Replaced by actual types used